internal/pkg/skuba/kubernetes/nodes_test.go

```diff
@@ -18,37 +18,409 @@
+									{
+										Name:  "test",
+										Image: "registry.suse.com/caasp/v4/test:1.1.1",
+									},
+								},
+							},
+						},
+					},
+				}
+				return true, obj, nil
+			})
+
+			if tt.evictSupport {
+				addEvictionSupport(t, fakeClientset, true)
+			}
+
+			node := corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: tt.node,
+				},
+			}
+			err := DrainNode(fakeClientset, &node, 10)
+
+			if tt.expectErrMsg != "" {
+				if err == nil {
+					t.Errorf("error expected on %s, but no error reported", tt.name)
+					return
+				}
+				if err.Error() != tt.expectErrMsg {
+					t.Errorf("returned error (%v) does not match the expected one (%v)", err.Error(), tt.expectErrMsg)
+					return
+				}
+				return
+			}
+			if tt.expectErrMsg == "" {
+				if err != nil {
+					t.Errorf("error not expected on %s, but an error was reported (%v)", tt.name, err.Error())
+					return
+				}
+
+				if tt.evictSupport {
+					var actualEviction []policyv1beta1.Eviction
+					for _, action := range fakeClientset.Actions() {
+						if action.GetVerb() != "create" ||
+							action.GetResource().Resource != "pods" ||
+							action.GetSubresource() != "eviction" {
+							continue
+						}
+
+						eviction := *action.(ktest.CreateAction).GetObject().(*policyv1beta1.Eviction)
+						actualEviction = append(actualEviction, eviction)
+					}
+					if len(actualEviction) == 0 {
+						t.Errorf("eviction expected on %s, but no eviction reported", tt.name)
+					}
+					return
+				}
+			}
+		})
+	}
+}
+
+func Test_getPodContainerImageTag(t *testing.T) {
+	fakeClientset := fake.NewSimpleClientset(
+		&corev1.NodeList{
+			Items: []corev1.Node{
+				m1, w1,
+			},
+		},
+		&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "valid",
+				Namespace: metav1.NamespaceDefault,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "test",
+						Image: "registry.suse.com/caasp/v4/test:1.1.1",
+					},
+				},
+			},
+		},
+	)
+
+	tests := []struct {
+		name         string
+		namespace    string
+		podName      string
+		expect       string
+		expectErrMsg string
+	}{
+		{
+			name:      "get pod container image tag",
+			namespace: metav1.NamespaceDefault,
+			podName:   "valid",
+			expect:    "1.1.1",
+		},
+		{
+			name:         "get pod container image tag with pod not exist",
+			namespace:    metav1.NamespaceDefault,
+			podName:      "invalid",
+			expectErrMsg: "could not retrieve pod object: pods \"invalid\" not found",
+		},
+		{
+			name:         "get pod container image tag with pod exist in different namespace",
+			namespace:    metav1.NamespaceSystem,
+			podName:      "invalid",
+			expectErrMsg: "could not retrieve pod object: pods \"invalid\" not found",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			actual, err := getPodContainerImageTag(fakeClientset, tt.namespace, tt.podName)
```

Comment:
Not about this PR. But it will be great to improve it.
If possible, it would better creates two files `pod.go` and `pod_test.go` and move functions `getPodContainerImageTag` and `Test_getPodContainerImageTag` into it :)

Reply:
Good suggestion.